client/src/ui/BitbakeTerminalLinkProvider.ts

```diff
@@ -0,0 +1,48 @@
+/* --------------------------------------------------------------------------------------------
+ * Copyright (c) 2023 Savoir-faire Linux. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ * ------------------------------------------------------------------------------------------ */
+
+import * as vscode from 'vscode'
+import { type BitBakeProjectScanner } from '../driver/BitBakeProjectScanner'
+import * as fs from 'fs'
+
+export class BitbakeTerminalLinkProvider implements vscode.TerminalLinkProvider {
+  bitBakeProjectScanner: BitBakeProjectScanner
+
+  constructor (bitBakeProjectScanner: BitBakeProjectScanner) {
+    this.bitBakeProjectScanner = bitBakeProjectScanner
+  }
+
+  async provideTerminalLinks (context: vscode.TerminalLinkContext, token: vscode.CancellationToken): Promise<vscode.TerminalLink[]> {
+    const links: vscode.TerminalLink[] = []
+    // We only have additional links to provide if we need to resolve container paths
+    if (!this.bitBakeProjectScanner.needsContainerPathsResolution()) { return links }
+
+    // Match any string that starts with a slash enclosed in spaces or special characters
+    const regex = /(^|[^a-zA-Z0-9_/.-])(\/[a-zA-Z0-9_/.-]+)(?=$|[^a-zA-Z0-9_/.-])/g
+    const matches = context.line.matchAll(regex)
+    for (const match of matches) {
+      if (match.index === undefined) continue
+      const link = new vscode.TerminalLink(
+        match.index + match[1].length,
```

Comment:
you could make the first group as "non capturing". This way the index would start at the part we're interested about, and it would make the code more straightforward

``` typescript
const regex = /(?:^|[^a-zA-Z0-9_/.-])(\/[a-zA-Z0-9_/.-]+)(?=$|[^a-zA-Z0-9_/.-])/g
```

```typescript
const link = new vscode.TerminalLink(
  match.index,
  match[1].length,
  ....
)
```

Reply:
It turns out if I use a non capturing group, I can't know the index of the initeresting one. I need the first group to be captured to know the length of the preceding special/blank character (always 1 in the current code, but still cleaner than hard coding).